### PR TITLE
FDX parser

### PR DESCRIPTION
**Pull request by bermannoah**
Closes #49

### CODE REVIEW

File: lib/parsers/fdx_parser.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  include FormatParser::IOUtils
+
+  def information_from_io(io)
+    if xml_check(io)
+      file_and_document_type = safe_read(io, 100)
+      file_type, document_type = check_for_document_type(file_and_document_type)
+    else
+      return # Bail if it's not even XML
+    end
+
```
Comment (julik) [inline lib/parsers/fdx_parser.rb]: Youa re having a bit of a gotofail here - if `check_for_document_type` returns a nil, you are going to return a `:document` FileInformation with a `nil` file_type, in practice you will be capturing _any_ XML document. Including, say, MusicXML and SVG.

File: lib/parsers/fdx_parser.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  def check_for_document_type(file_and_document_type)
+    sanitized_data = file_and_document_type.downcase
+    if sanitized_data.include?("finaldraft") && sanitized_data.include?("script") 
+      return :fdx, :script
+    else
+      return
+    end
+  end
+
+end
```
Comment (julik) [inline lib/parsers/fdx_parser.rb]: Newline

File: spec/parsers/fdx_parser_spec.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    Dir.glob(fixtures_dir + '/*.fdx').each do |fdx_path|
+      it "is able to parse #{File.basename(fdx_path)}" do
+        parsed = subject.information_from_io(File.open(fdx_path, 'rb'))
+        expect(parsed).not_to be_nil
+        expect(parsed.file_nature).to eq(:document)
+        expect(parsed.file_type).to eq(:fdx)
+        expect(parsed.document_type).to eq(:script)
+      end
+    end
+  end
```
Comment (julik) [inline spec/parsers/fdx_parser_spec.rb]: You could use a test here that you are not intercepting other XML documents by accident (right now you do).

Comment (bermannoah) [reply spec/parsers/fdx_parser_spec.rb]: 👍

**Comment by bermannoah** [comment]
Unfortunately but not unsurprisingly, the FDX files don't contain a "total number of pages" property. They _do_ contain a list of system voices to potentially use for each character, but no TotalPages info. This probably explains something but I can't think what.

**Comment by julik** [comment]
> This probably explains something but I can't think what.

That one is easy - that they want you to buy FD to be able to compute the number of screen minutes, what else.

**Comment by bermannoah** [comment]
"It's a feature, not a bug"

File: lib/parsers/fdx_parser.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    sanitized_data = file_and_document_type.downcase
+    if sanitized_data.include?("finaldraft") && sanitized_data.include?("script") 
+      return :fdx, :script
+    else
+      return
+    end
+  end
+
+  FormatParser.register_parser_constructor self
+end
```
Comment (julik) [inline lib/parsers/fdx_parser.rb]: Still a newline

**Pull request merged**